grammar.js

```diff
@@ -34,7 +35,55 @@ module.exports = grammar({
     as_name: $ => seq(
       'as',
       $.identifier
-    )
+    ),
+
+    assign_definition: $ => choice(
+      $.assign,
+      $.augmented_assign,
+    ),
+
+    // TODO(wei): should be typed
+    test: _ => /[a-zA-Z0-9."']+/,
+
+    assign: $ => seq(
+      choice(
+        $.identifier,
+        seq(
+          $.identifier,
+          repeat(seq(DOT, $.identifier)),
+        ),
+      ),
+      '=',
+      $.test,
+    ),
+
+    augassign: _ => choice("+=",
+      "-=",
+      "*=",
+      "**=",
+      "/=",
+      "//=",
+      "%=",
+      "&=",
+      "|=",
+      "^=",
+      "<<=",
+      ">>=",
+      "or",
```

Comment:
`or` and `and` may not be right aug assign operators. And the operator precedence document is here: https://kcl-lang.io/docs/reference/lang/spec/kcl-spec/#operator-precedence

There may mistakes for specific grammar spec. U can ref https://kcl-lang.io/docs/reference/lang/spec/kcl-spec#grammar

Reply:
Oops, let me fix it but then it seems that there may be some mistakes [here](https://kcl-lang.io/docs/reference/lang/spec/statements/#assignment-statements)? as I copied the list from there to save my time...

<img width="900" alt="image" src="https://github.com/kcl-lang/tree-sitter-kcl/assets/3764335/f9c3efa7-1039-4154-9650-e6389ec03030">

Reply:
> Oops, let me fix it but then it seems that there may be some mistakes [here](https://kcl-lang.io/docs/reference/lang/spec/statements/#assignment-statements)? as I copied the list from there to save my time...
> 
> <img alt="image" width="900" src="https://user-images.githubusercontent.com/3764335/270096715-f9c3efa7-1039-4154-9650-e6389ec03030.png">

Yes. I'll fix and double check them. U can ref the total grammar: https://kcl-lang.io/docs/reference/lang/spec/kcl-spec#grammar